### ISSUE TITLE
docker-in-docker dind on Apple Silicon

### ISSUE THREAD

**Issue by justinTM** [firecow/gitlab-ci-local#1035]
**Is your feature request related to a problem? Please describe.**
I get this error about mismatching architectures when trying to enable docker-in-docker dind:
```console
(devbox) bash-3.2$  gitlab-ci-local "Apply Dev" --variable CI_PROJECT_ID=$CI_PROJECT_ID --variable CI_JOB_TOKEN=$CI_JOB_TOKEN  --volume ~/.aws:/root/.aws --privileged --ulimit=8000:16000 --variable DOCKER_TLS_CERTDIR=/tmp/kafka/kafka.client.truststore.jks
parsing and downloads finished in 323 ms
Apply Dev       starting registry.gitlab.com/----/pipelines:----move-to-gitlab (Stage)
Apply Dev       copied to docker volumes in 867 ms
Apply Dev       started service image: docker:dind with aliases: docker in 411 ms
Apply Dev       > still running...
Apply Dev       > still running...
Apply Dev       > still running...
Apply Dev       service image: docker:dind healthcheck failed
Apply Dev         Command failed with exit code 124: docker run --rm --name=gcl-wait-for-it-126041-0-2375 --network gitlab-ci-local-126041 docker.io/willwill/wait-for-it docker:2375 -t 30
Apply Dev         Info: Your devbox environment may be out of date. Please run 
Apply Dev         
Apply Dev         eval "$(devbox global shellenv --recompute)"
Apply Dev         
Apply Dev         WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Apply Dev         wait-for-it.sh: waiting 30 seconds for docker:2375
Apply Dev         wait-for-it.sh: timeout occurred after waiting 30 seconds for docker:2375
Apply Dev       also see (/Users/justin/git/---/lambdas/regex/.gitlab-ci-local/services-output/ApplyIADev/docker:dind-0.log)
Apply Dev       service image: docker:dind healthcheck failed
Apply Dev         Command failed with exit code 124: docker run --rm --name=gcl-wait-for-it-126041-0-2376 --network gitlab-ci-local-126041 docker.io/willwill/wait-for-it docker:2376 -t 30
Apply Dev         Info: Your devbox environment may be out of date. Please run 
Apply Dev         
Apply Dev         eval "$(devbox global shellenv --recompute)"
Apply Dev         
Apply Dev         WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Apply Dev         wait-for-it.sh: waiting 30 seconds for docker:2376
Apply Dev         wait-for-it.sh: timeout occurred after waiting 30 seconds for docker:2376
Apply Dev       also see (/Users/justin/git/---/lambdas/regex/.gitlab-ci-local/services-output/ApplyIADev/docker:dind-0.log)
```

**Describe the solution you'd like**
I'd love a way to specify architecture for the required images (apparently `docker.io/willwill/wait-for-it`?)

**Describe alternatives you've considered**

**Additional context**

```console
$ gitlab-ci-local --version
4.45.1
Apply Dev       $ docker --version
Apply Dev       > Docker version 19.03.15, build 99e3ed8
```

**Comment by lisn0** [comment]
Issue resolved with #1033 .

**Comment by justinTM** [comment]
@lisn0 awesome thank you so much!
